algorithms/src/sorting/impl/Kokkos_SortImpl.hpp

```diff
@@ -206,10 +209,49 @@ void sort_onedpl(const Kokkos::Experimental::SYCL& space,
   auto queue  = space.sycl_queue();
   auto policy = oneapi::dpl::execution::make_device_policy(queue);
   const int n = view.extent(0);
-  oneapi::dpl::sort(policy, view.data(), view.data() + n);
+  oneapi::dpl::sort(policy, view.data(), view.data() + n,
+                    std::forward<MaybeComparator>(maybeComparator)...);
 }
 #endif
 
+template <class ExecutionSpace, class DataType, class... Properties,
+          class... MaybeComparator>
+void copy_to_host_run_stdsort_copy_back(
+    const ExecutionSpace& exec,
+    const Kokkos::View<DataType, Properties...>& view,
+    MaybeComparator&&... maybeComparator) {
+  namespace KE = ::Kokkos::Experimental;
+
+  using ViewType = Kokkos::View<DataType, Properties...>;
+  using layout   = typename ViewType::array_layout;
+  if constexpr (std::is_same_v<LayoutStride, layout>) {
+    // for strided views we cannot just deep_copy from device to host,
+    // so we need to do a few more jumps
+    using view_value_type      = typename ViewType::non_const_value_type;
+    using view_exespace        = typename ViewType::execution_space;
+    using view_deep_copyable_t = Kokkos::View<view_value_type*, view_exespace>;
+    view_deep_copyable_t view_dc("view_dc", view.extent(0));
+    KE::copy(exec, view, view_dc);
+    exec.fence();
```

Comment:
Why fence?

Reply:
i dont need it here?

Reply:
The create_mirror_ciew_and_copy below will sync but my point was more you need to label the fence.

Reply:
i removed the fence, we don't actually need it because KE::copy fences already